wurst/file/Base64.wurst

```diff
@@ -0,0 +1,270 @@
+package Base64
+
+import Bitwise
+import StringUtils
+import ByteBuffer
+import ChunkedString
+import Execute
+import ErrorHandling
+
+/*  Base64 package provides an encoder/decoder pair that allows serializing and deserializing of any data.
+
+    Encode Data iteratively:
+    > let encoder = new Base64Encoder()
+    > encoder.writeByte(25)
+    > encoder.writeInt(100500)
+    > let data = encoder.intoData() // the encoder is consumed, no need to destroy it
+
+    Decode Data iteratively:
+    > let decoder = new Base64Decoder()
+    > while data.hasChunk()
+    >   decoder.append(data.readChunk())
+    > let bytes = decoder.intoData() // the decoder is consumed, no need to destroy it
+    > let byte = bytes.readByte()
+    > let number = bytes.readInt()
```

Comment:
This API block is a bit bizarre IMO. Is `decoder` a mutable, stateful iterator for `data`?

Reply:
No, the decoder only accepts strings to be decoded and returns a `ByteBuffer` with data.